dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/python/file/type/ZipFileType.java

```diff
@@ -0,0 +1,75 @@
+package io.odpf.dagger.functions.udfs.python.file.type;
+
+import io.odpf.dagger.functions.udfs.python.file.source.FileSource;
+import io.odpf.dagger.functions.udfs.python.file.source.FileSourceFactory;
+import lombok.SneakyThrows;
+
+import java.io.ByteArrayInputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+/**
+ * The type Zip file type.
+ */
+public class ZipFileType implements FileType {
+
+    private String pythonFile;
+    private FileSourceFactory fileSourceFactory;
+
+    /**
+     * Instantiates a new Zip file type.
+     *
+     * @param pythonFile the python file
+     */
+    public ZipFileType(String pythonFile) {
+        this.pythonFile = pythonFile;
+    }
+
+    /**
+     * Instantiates a new Zip file type.
+     * This constructor used for unit test purposes.
+     *
+     * @param pythonFile  the python file
+     * @param fileSourceFactory the file source factory
+     */
+    public ZipFileType(String pythonFile, FileSourceFactory fileSourceFactory) {
+        this.pythonFile = pythonFile;
+        this.fileSourceFactory = fileSourceFactory;
+    }
+
+    @SneakyThrows
```

Comment:
the interface of this class didn't contain `throws IOException`. this is because we are using a method that requires this. there are few options for this actually.
1. adding SneakyThrows
2. add throws IOException on the interface, but PythonFileType will also have this which is not necessary
3. add try-catch inside the method. (personally, I don't like using this :p )

which one do you think we should use?